File: Colage/Features/Auth/Screens/PhotoUploadScreen.swift
import SwiftUI
import PhotosUI

struct PhotoUploadScreen: View {
    let onContinue: () -> Void
    @State private var selectedItem: PhotosPickerItem?
    @State private var profileImage: Image?
    @State private var profileUIImage: UIImage?
    @State private var showCamera = false
    @State private var capturedImage: UIImage?

    var body: some View {
        ZStack {
            ColageColors.background.ignoresSafeArea()

            VStack(spacing: 0) {
                OnboardingProgress(currentStep: 4, totalSteps: 10)
                    .padding(.top, 8)

                VStack(spacing: 12) {
                    Text("Add a profile photo")
                        .font(ColageFonts.title)
                        .foregroundStyle(ColageColors.textPrimary)
                    Text("This is how people will find you")
                        .font(ColageFonts.body)
                        .foregroundStyle(ColageColors.textSecondary)
                }
                .padding(.top, 48)

                Spacer().frame(height: 48)

                // Photo preview
                ZStack {
                    if let profileImage {
                        profileImage
                            .resizable()
                            .scaledToFill()
                            .frame(width: 160, height: 160)
                            .clipShape(Circle())
                            .overlay(
                                Circle()
                                    .strokeBorder(ColageColors.primary, lineWidth: 3)
                            )
                    } else {
                        Circle()
                            .fill(ColageColors.surface)
                            .frame(width: 160, height: 160)
                            .overlay(
                                VStack(spacing: 8) {
                                    Image(systemName: "camera.fill")
                                        .font(.system(size: 36))
                                    Text("Add Photo")
                                        .font(ColageFonts.caption)
                                }
                                .foregroundStyle(ColageColors.textTertiary)
                            )
                            .overlay(
                                Circle()
                                    .strokeBorder(ColageColors.border, lineWidth: 2)
                            )
                    }

                    // Change photo button (when photo exists)
                    if profileImage != nil {
                        VStack {
                            Spacer()
                            HStack {
                                Spacer()
                                Image(systemName: "pencil.circle.fill")
                                    .font(.system(size: 32))
                                    .foregroundStyle(ColageColors.primary)
                                    .background(Circle().fill(ColageColors.background).frame(width: 28, height: 28))
                            }
                        }
                        .frame(width: 160, height: 160)
                    }
                }

                Spacer().frame(height: 32)

                // Photo source buttons
                HStack(spacing: 16) {
                    PhotosPicker(selection: $selectedItem, matching: .images) {
                        Label("Library", systemImage: "photo.on.rectangle")
                            .font(ColageFonts.bodyBold)
                            .foregroundStyle(ColageColors.primary)
                            .frame(maxWidth: .infinity)
                            .frame(height: 48)
                            .background(ColageColors.primary.opacity(0.12))
                            .clipShape(RoundedRectangle(cornerRadius: 12))
                    }

                    Button {
                        showCamera = true
                    } label: {
                        Label("Camera", systemImage: "camera")
                            .font(ColageFonts.bodyBold)
                            .foregroundStyle(ColageColors.primary)
                            .frame(maxWidth: .infinity)
                            .frame(height: 48)
                            .background(ColageColors.primary.opacity(0.12))
                            .clipShape(RoundedRectangle(cornerRadius: 12))
                    }
                }
                .padding(.horizontal, 24)

                Spacer()

                VStack(spacing: 12) {
                    ColagePrimaryButton(title: "Continue", action: onContinue)

                    Button("Skip for now") {
                        onContinue()
                    }
                    .font(ColageFonts.body)
                    .foregroundStyle(ColageColors.textSecondary)
                }
                .padding(.horizontal, 24)
                .padding(.bottom, 40)
            }
        }
        .fullScreenCover(isPresented: $showCamera) {
            CameraCaptureView(capturedImage: $capturedImage)
                .ignoresSafeArea()
        }
        .onChange(of: selectedItem) { _, newItem in
            Task {
                if let data = try? await newItem?.loadTransferable(type: Data.self),
                   let uiImage = UIImage(data: data) {
                    profileUIImage = uiImage
                    profileImage = Image(uiImage: uiImage)
                }
            }
        }
        .onChange(of: capturedImage) { _, newImage in
            if let newImage {
                profileUIImage = newImage
                profileImage = Image(uiImage: newImage)
            }
        }
    }
}

#Preview {
    NavigationStack {
        PhotoUploadScreen(onContinue: {})
    }
}

File: android/app/src/main/java/com/colageclub/colage/features/auth/screens/PhotoUploadScreen.kt
package com.colageclub.colage.features.auth.screens

import android.net.Uri
import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.PickVisualMediaRequest
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CameraAlt
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material.icons.filled.Photo
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import coil.compose.AsyncImage
import com.colageclub.colage.core.design.*
import com.colageclub.colage.features.auth.AuthViewModel

@Composable
fun PhotoUploadScreen(
    authViewModel: AuthViewModel,
    onContinue: () -> Unit
) {
    val onboardingData by authViewModel.onboardingData.collectAsState()
    var selectedUri by remember { mutableStateOf<Uri?>(onboardingData.profilePhotoUri) }

    val photoPicker = rememberLauncherForActivityResult(
        ActivityResultContracts.PickVisualMedia()
    ) { uri ->
        uri?.let {
            selectedUri = it
            authViewModel.updateOnboardingPhoto(it)
        }
    }

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(ColageColors.Background)
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .systemBarsPadding(),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            OnboardingProgress(currentStep = 4, totalSteps = 10, modifier = Modifier.padding(top = 8.dp))

            Column(
                modifier = Modifier.padding(top = 48.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                Text(
                    text = "Add a profile photo",
                    style = ColageFonts.Title.copy(color = ColageColors.TextPrimary)
                )
                Text(
                    text = "This is how people will find you",
                    style = ColageFonts.Body.copy(color = ColageColors.TextSecondary)
                )
            }

            Spacer(Modifier.height(48.dp))

            // Photo preview
            Box(modifier = Modifier.size(160.dp)) {
                if (selectedUri != null) {
                    AsyncImage(
                        model = selectedUri,
                        contentDescription = "Profile photo",
                        contentScale = ContentScale.Crop,
                        modifier = Modifier
                            .fillMaxSize()
                            .clip(CircleShape)
                            .border(3.dp, ColageColors.Primary, CircleShape)
                    )
                    // Edit button overlay
                    Box(
                        modifier = Modifier
                            .align(Alignment.BottomEnd)
                            .size(36.dp)
                            .clip(CircleShape)
                            .background(ColageColors.Primary),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = Icons.Default.Edit,
                            contentDescription = "Change photo",
                            tint = ColageColors.TextPrimary,
                            modifier = Modifier.size(18.dp)
                        )
                    }
                } else {
                    Box(
                        modifier = Modifier
                            .fillMaxSize()
                            .clip(CircleShape)
                            .background(ColageColors.Surface)
                            .border(2.dp, ColageColors.Border, CircleShape),
                        contentAlignment = Alignment.Center
                    ) {
                        Column(
                            horizontalAlignment = Alignment.CenterHorizontally,
                            verticalArrangement = Arrangement.spacedBy(8.dp)
                        ) {
                            Icon(
                                imageVector = Icons.Default.CameraAlt,
                                contentDescription = null,
                                tint = ColageColors.TextTertiary,
                                modifier = Modifier.size(36.dp)
                            )
                            Text(
                                text = "Add Photo",
                                style = ColageFonts.Caption.copy(color = ColageColors.TextTertiary)
                            )
                        }
                    }
                }
            }

            Spacer(Modifier.height(32.dp))

            // Source buttons
            Row(
                modifier = Modifier.padding(horizontal = 24.dp),
                horizontalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                OutlinedButton(
                    onClick = {
                        photoPicker.launch(
                            PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly)
                        )
                    },
                    modifier = Modifier
                        .weight(1f)
                        .height(48.dp),
                    shape = RoundedCornerShape(12.dp),
                    colors = ButtonDefaults.outlinedButtonColors(
                        contentColor = ColageColors.Primary,
                        containerColor = ColageColors.Primary.copy(alpha = 0.12f)
                    ),
                    border = null
                ) {
                    Icon(Icons.Default.Photo, contentDescription = null, modifier = Modifier.size(18.dp))
                    Spacer(Modifier.width(8.dp))
                    Text("Library", style = ColageFonts.BodyBold)
                }

                OutlinedButton(
                    onClick = {
                        photoPicker.launch(
                            PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly)
                        )
                    },
                    modifier = Modifier
                        .weight(1f)
                        .height(48.dp),
                    shape = RoundedCornerShape(12.dp),
                    colors = ButtonDefaults.outlinedButtonColors(
                        contentColor = ColageColors.Primary,
                        containerColor = ColageColors.Primary.copy(alpha = 0.12f)
                    ),
                    border = null
                ) {
                    Icon(Icons.Default.CameraAlt, contentDescription = null, modifier = Modifier.size(18.dp))
                    Spacer(Modifier.width(8.dp))
                    Text("Camera", style = ColageFonts.BodyBold)
                }
            }

            Spacer(Modifier.weight(1f))

            Column(
                modifier = Modifier.padding(horizontal = 24.dp, vertical = 40.dp),
                verticalArrangement = Arrangement.spacedBy(12.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                ColagePrimaryButton(
                    title = "Continue",
                    onClick = onContinue
                )

                TextButton(onClick = onContinue) {
                    Text(
                        text = "Skip for now",
                        style = ColageFonts.Body.copy(color = ColageColors.TextSecondary)
                    )
                }
            }
        }
    }
}
